.github/workflows/build_steps.yml

```diff
@@ -0,0 +1,176 @@
+name: build_steps
+on:
+  workflow_call:
+    inputs:
+      job_type:          {required: true, type: string, description: Usually caller job's name}
+      cmake_preset_type: {required: true, type: string, description: See the synonyms param in parent workflow}
+      cibw_image_tag:    {required: true, type: string, description: As build by cibw_docker_image.yml workflow}
+      python:            {default: '(default)', type: string, description: Python version for followers}
+
+jobs:
+  compile:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [linux]
+        include:
+          - os: linux
+            distro: ubuntu-latest
+            cmake_preset_prefix: linux
+            cibw_format: "cp{0}-manylinux_x86_64"
+          # Windows is not included ATM because the build would run out of disk space ¯\_(ツ)_/¯
+          #- os: windows
+          #  distro: windows-latest
+          #  cmake_preset_prefix: windows-cl
+
+    name: build on ${{matrix.os}}
+    runs-on: ${{matrix.distro}}
+    container: ${{ (matrix.os == 'linux' && inputs.job_type != 'follower') && inputs.cibw_image_tag || null}}
+    env:
+      SCCACHE_GHA_VERSION: ${{vars.SCCACHE_GHA_VERSION || 1}} # Setting this env var enables the caching
+      CMAKE_C_COMPILER_LAUNCHER: sccache
+      CMAKE_CXX_COMPILER_LAUNCHER: sccache
+      ARCTIC_CMAKE_PRESET: ${{matrix.cmake_preset_prefix}}-${{inputs.cmake_preset_type}}
+      VCPKG_BINARY_SOURCES: 'clear;nuget,github,readwrite'
+      nuget_user: ${{vars.vcpkg_nuget_user || github.repository_owner}}
+      nuget_token: ${{vars.vcpkg_nuget_token || secrets.GITHUB_TOKEN}}
+      CIBW_ENVIRONMENT_PASS_LINUX: SCCACHE_GHA_VERSION ACTIONS_CACHE_URL ACTIONS_RUNTIME_TOKEN
+        CMAKE_C_COMPILER_LAUNCHER CMAKE_CXX_COMPILER_LAUNCHER CMAKE_BUILD_PARALLEL_LEVEL
+        ARCTIC_CMAKE_PRESET VCPKG_BINARY_SOURCES nuget_user nuget_token test_output
+      CIBW_BUILD: ${{format(matrix.cibw_format, inputs.python)}}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3.3.0
+        with:
+          submodules: recursive # Just in case a dep has its own third-party deps
+
+      - name: Deep checkout vcpkg
+        run: cd cpp/vcpkg ; git fetch --unshallow origin master
+
+      - name: Configure sccache
```

Comment:
Can use it locally by pointing the `CMAKE_*_COMPILER_LAUNCHER` to it.

It's also a remote/distributed compiler luncher at the same time and I am hoping of setting up at a later date:
* A shared cache interally (cannot easily reuse the GitHub one)
* A set of distributed worker nodes (instead of everyone trying to get larger headnodes)